src/calendar/MonthPicker.ts

```diff
@@ -0,0 +1,311 @@
+			const { open } = this.properties;
+			if (open && this._callPopupFocus) {
+				element.focus();
+				this._callPopupFocus = false;
+			}
+		}
+	}
+
+	private _closePopup() {
+		const { onRequestClose } = this.properties;
+
+		this._callTriggerFocus = true;
+		onRequestClose && onRequestClose();
+	}
+
+	private _openPopup() {
+		const { onRequestOpen } = this.properties;
+
+		this._callPopupFocus = true;
+		onRequestOpen && onRequestOpen();
+	}
+
+	private _decreaseYear() {
+		const {
+			year,
+			onRequestYearChange
+		} = this.properties;
+
+		onRequestYearChange && onRequestYearChange(year - 1);
+	}
+
+	private _increaseYear() {
+		const {
+			year,
+			onRequestYearChange
+		} = this.properties;
+
+		onRequestYearChange && onRequestYearChange(year + 1);
+	}
+
+	private _onButtonClick() {
+		const {
+			open = false
+		} = this.properties;
+
+		open ? this._closePopup() : this._openPopup();
+	}
+
+	private _onMonthDecrease() {
+		const {
+			month,
+			year,
+			onRequestMonthChange,
+			onRequestYearChange
+		} = this.properties;
+
+		if (month === 0) {
+			onRequestMonthChange && onRequestMonthChange(11);
+			onRequestYearChange && onRequestYearChange(year - 1);
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month - 1);
+		}
+	}
+
+	private _onMonthIncrease() {
+		const {
+			month,
+			year,
+			onRequestMonthChange,
+			onRequestYearChange
+		} = this.properties;
+
+		if (month === 11) {
+			onRequestMonthChange && onRequestMonthChange(0);
+			onRequestYearChange && onRequestYearChange(year + 1);
+		}
+		else {
+			onRequestMonthChange && onRequestMonthChange(month + 1);
+		}
+	}
+
+	private _onPopupKeyDown(event: KeyboardEvent) {
+		// close popup on escape, or if a value is selected with enter/space
+		if (
+			event.which === Keys.Escape ||
+			event.which === Keys.Enter ||
+			event.which === Keys.Space
+		) {
+			this._closePopup();
+		}
+	}
+
+	private _onRadioChange(event: Event) {
+		const {
+			onRequestMonthChange
+		} = this.properties;
+		const month = parseInt((<HTMLInputElement> event.target).value, 10);
+
+		onRequestMonthChange && onRequestMonthChange(month);
+	}
+
+	private _onYearKeyDown(event: KeyboardEvent) {
+		switch (event.which) {
+			case Keys.Right:
+				this._increaseYear();
+				break;
+			case Keys.Left:
+				this._decreaseYear();
+				break;
+		}
+	}
+
+	private _renderMonthLabel(month: number, year: number) {
+		const { monthNames, renderMonthLabel } = this.properties;
+		return renderMonthLabel ? renderMonthLabel(month, year) : monthNames[month].long + ' ' + year;
+	}
+
+	private _renderMonthRadios() {
+		const { month, theme = {} } = this.properties;
+
+		return this.properties.monthNames.map((monthName, i) => w(Radio, {
+			key: this._radiosName + i,
+			overrideClasses: { root: css.monthRadio, input: css.monthRadioInput, checked: css.monthRadioChecked },
+			checked: i === month,
+			label: {
+				content: `<abbr title="${monthName.long}">${monthName.short}</abbr>`,
```

Comment:
That's an interesting question. It's currently typed as `content: string` in form widget properties, and I'm not sure I'd want to encourage adding DNodes to label content. It could be changed, though.